.pre-commit-config.yaml

```
@@ -0,0 +1,82 @@
+# ci:
+#   autofix_commit_msg: "ci(pre-commit): autofix"
+#   autoupdate_commit_msg: "ci(pre-commit): autoupdate"
+
+repos:
+  # - repo: https://github.com/pre-commit/pre-commit-hooks
+  #   rev: v4.1.0
+  #   hooks:
+  #     - id: check-json
+  #     - id: check-merge-conflict
+  #     - id: check-toml
+  #     - id: check-xml
+  #     - id: check-yaml
+  #       args: [--unsafe]
+  #     - id: detect-private-key
+  #     - id: end-of-file-fixer
+  #     - id: mixed-line-ending
+  #     - id: trailing-whitespace
+  #       args: [--markdown-linebreak-ext=md]
+
+  # - repo: https://github.com/igorshubovych/markdownlint-cli
+  #   rev: v0.31.1
+  #   hooks:
+  #     - id: markdownlint
+  #       args: [-c, .markdownlint.yaml, --fix]
+
+  # - repo: https://github.com/pre-commit/mirrors-prettier
+  #   rev: v2.6.2
+  #   hooks:
+  #     - id: prettier
+
+  # - repo: https://github.com/adrienverge/yamllint
+  #   rev: v1.26.3
+  #   hooks:
+  #     - id: yamllint
+
+  - repo: https://github.com/tier4/pre-commit-hooks-ros
+    rev: v0.8.0
+    hooks:
+      - id: flake8-ros
+  #     - id: prettier-xacro
+  #     - id: prettier-launch-xml
+  #     - id: prettier-package-xml
+  #     - id: sort-package-xml
+  #     - id: ros-include-guard
+
+  # - repo: https://github.com/shellcheck-py/shellcheck-py
+  #   rev: v0.8.0.4
+  #   hooks:
+  #     - id: shellcheck
+
+  # - repo: https://github.com/scop/pre-commit-shfmt
+  #   rev: v3.4.3-1
+  #   hooks:
+  #     - id: shfmt
+  #       args: [-w, -s, -i=4]
+
+  # - repo: https://github.com/pycqa/isort
+  #   rev: 5.12.0
+  #   hooks:
+  #     - id: isort
+
+  # - repo: https://github.com/psf/black
+  #   rev: 22.3.0
+  #   hooks:
+  #     - id: black
+  #       args: [--line-length=100]
+
+  # - repo: https://github.com/pre-commit/mirrors-clang-format
+  #   rev: v13.0.1
+  #   hooks:
+  #     - id: clang-format
+  #       types_or: [c++, c, cuda]
+
+  # - repo: https://github.com/cpplint/cpplint
+  #   rev: 1.6.0
+  #   hooks:
+  #     - id: cpplint
+  #       args: [--quiet]
+  #       exclude: .cu
+
+exclude: .svg|control/trajectory_follower|control/trajectory_follower_nodes|common/autoware_auto_cmake|common/autoware_auto_common|common/autoware_auto_geometry|common/autoware_auto_tf2|common/autoware_testing|common/fake_test_node|common/had_map_utils|common/motion_common|common/motion_testing|common/time_utils|common/vehicle_constants_manager|common/autoware_auto_perception_rviz_plugin|common/osqp_interface|simulator/simple_planning_simulator|planning/freespace_planner|planning/astar_search|planning/costmap_generator
```

```suggestion
exclude: .svg
```
この設定はautoware.universe設定の名残かと思います。v2i_interfaceにこれらのディレクトリはないため、残すとしてもsvgファイルの除外設定のみでよいのではないでしょうか？

---

@yn-mrse 
svgファイルの除外設定のみにしたいと思います！

---

@yn-mrse 
https://github.com/eve-autonomy/v2i_interface/pull/59/commits/e522c9e1b1e95417ee809c81410ff884553c36c6
で対応しました！